platform/engineering/design-docs/2022-01-12-Elixir.md

```diff
@@ -0,0 +1,100 @@
+
+### Debugging
+  Use IEx.pry to interact with any component the same as you would Ruby pry. Debugging in Elixir is no different than debugging a rails app, so for the purposes of this document we will not expand further at this time.
+
+### Caveats
+_To be determined._
+
+### Security Concerns
+  Elixir and the phoenix framework will be deployed as services within our current K8s va.gov infrastructure. There are no additional concerns that are worth noting at this time that are outside of what you would expect when creating an application similar to ruby on rails. In fact Elixirs share nothing concurrency makes this language more secure than many others, including Ruby.
+
+### Privacy Concerns
+Privacy concerns are not a direct consideration of the implementation of Elixir within the sign in service.
+
+### Open Questions and Risks
+- Are we concerned with the lack of current Elixir knowledge within the VA?
+  - Generally we are not because the language is not more complex than Ruby and so similar to Ruby that many of the current Ruby engineers will be able to pick this language up without any issues. The team also intends on creating a community of practice which will provide other teams support with considering and implementing Elixir within any new solutions they have.
+  - Likelihood: Moderate
+  - Cost: hours spent to build this service would be lost if it was determined that support of the Elixir application was not sustainable
+  - Mitigations: 
+    - Initial service will be very small (less than 1k loc)
+    - Hold Elixir workshops and establish CoP
+- What if the response and concurrency do not deliver as expected?
+  - We will document the performance of the implemented service and determine if we should continue with Elixir or pivot to another language.
+  - Likelihood: Low
+  - Cost: hours spent to build this service would be lost if it was determined that support of the Elixir application was not sustainable
+  - Mitigations:
+    - Reduction of likelihood by measuring the performance throughout the development cycle 
+
+### Work Estimates
+The phases of work are as follows:
+- The use of Elixir will come with gaining knowledge of the language and its framework Phoenix
+  - 4 hours per current Ruby engineer
+  - 2 hours per current security engineer
+  - 4 hours per supporting engineer (operations, testing) 
+- Implement the first service with this language (sign in service), includes initial architecture integration
+  - 4 sprints
+- Establish CoP for Elixir
+  - 1 sprint for initial creation, varies for maintenance and office hours after
+- Implement additional services with Elixir as appropriate
+  - Varies based on complexity of the current service and solution, however the scaffolding and examples of other applications
+
+### Alternatives
+If Elixir does not deliver on the expected advantages we can either pivot to Go or Ruby. 
```

Comment:
It would helpful to see the unique advantages of Elixir OTP/Phoenix framework listed out in a section versus the other alternatives of (Ruby, Nodejs, and Golang) (*Completely optional)

Reply:
I added more content here but I do welcome anyone wanting to add more to compare phoenix to rails